packages/core/src/contrib/Functions.ts

```diff
@@ -211,6 +211,20 @@ export const compDict = {
     };
   },
 
+  /**
+   * Return a paint of none (no paint)
+   */
+  none: (): IColorV<VarAD> => {
+    return {
+      tag: "ColorV",
+      contents: {
+        tag: "NONE",
+        // Irrelevant dummy color -- paint=none is a special case in the renderer
```

Comment:
Is the dummy color needed here? Maybe the type `INone` shouldn't have `contents` in the first place?

Reply:
Also, I think `variableAD` signals to the autodiff engine that this is a variable, not a constant. `constOf(0)` is probably the better function to call if you want to create a constant.

Reply:
Tend to agree: the color quad is vestigial in this case as the renderer will ignore it.  

I was similarly tempted to remove this and have an alternate local implementation to that effect, but it carries some downsides.  As we implement more aspects of SVG paint, we will have more "paint" information flowing through the middle of the system -- image and gradient paints might be future examples -- and it's a valid question about how closely we need to manage that data structure before it gets to the renderer. I.e., do we only include the relevant data in the structure (i.e., the structure for color, none, image, etc. are all different) or do we put the paint information we have in one structure and let the renderer decide what it needs, probably based on the tag.

After thinking through both implementations, it seemed simpler (short and long-term) to keep the interface structure consistent as a "paint" structure; varying it by tag triggers a fair amount of if/then/else and casting in the middle of the pipeline to deal with the structure variances based on what the type checker flagged.  I know a couple places where color/paint is typed 'any,' which leads me to think there might be other places that need to be altered to avoid runtime errors the static checker couldn't find automatically.  

The net effect either way seems to be the same: the renderer will have to decide how to apply paint at the end of the pipeline, probably based on the tag plus whichever fields in the structure it decides it needs.  Varying the structure of the type before it gets to the renderer, on balance, seemed to generate more code just to the structural difference without bringing us much benefit short or long term.  

I still have that alternate implementation and can do some more testing to try and tease out any runtime exceptions and then send it as a PR -- I am fine with that if it is the best thing for the project.  With either implementation, we can always feel free to change our minds later.